src/apps/compare.C

```diff
@@ -152,10 +148,10 @@ void process_cmd_line(int argc,
            */
         case 'r':
           {
-            if ((!names.empty()) && (left_name_set))
-              names.push_back(optarg);
-            else
-              libmesh_error_msg("ERROR: Mesh file name and left file name must precede right file name!");
+            libmesh_error_msg_if(names.empty() || !left_name_set,
```

Comment:
> [I don't trust myself](https://github.com/libMesh/libmesh/pull/2660#issuecomment-664539859) to DeMorgan-ize complicated compound logical expressions

I thought that line was a good explanation for your decision; now I fear it was also foreshadowing...

Reply:
Not gonna lie, I have the [wikipedia page](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) open in one of my tabs right now.